### PR TITLE
Add additional docstrings

### PR DESCRIPTION
**Pull request by michel-laterman**
Add additional docstrings

### CODE REVIEW

File: internal/pkg/rollback/rollback.go [hunk 1]
```diff
@@ -11,30 +11,34 @@ import (
...
 
+// RollbackFunc is a function that is called in order to perform a rollback.
 type RollbackFunc func(ctx context.Context) error
 
 type rollbackInfo struct {
 	name string
 	fn   RollbackFunc
 }
 
+// Rollback is used to track RollbackFuncs
```
Comment (aleksmaus) [inline internal/pkg/rollback/rollback.go:22]: dot "." at the end? you seemed to be adding these above

File: internal/pkg/rollback/rollback.go [hunk 2]
```diff
@@ -11,30 +11,34 @@ import (
...
 }
 
+// New returns a new Rollback.
 func New(log zerolog.Logger) *Rollback {
 	return &Rollback{
 		log: log,
 	}
 }
 
+// Register adds the named function to Rollback
```
Comment (aleksmaus) [inline internal/pkg/rollback/rollback.go:35]: dot?

File: internal/pkg/scheduler/scheduler.go [hunk 3]
```diff
@@ -78,6 +87,9 @@ func New(schedules []Schedule, opts ...OptFunc) (*Scheduler, error) {
 	return s, nil
 }
 
+// Run executes all scheduled function according to their schedules.
+// Schedule Interval times are garunteed minium values (if a execution takes a very long time, the scheduler will wait Interval before running the function again).
```
Comment (aleksmaus) [inline internal/pkg/scheduler/scheduler.go:91]: garunteed -> guaranteed

"if a execution takes a very long time" -> "if the execution takes a very long time"

File: internal/pkg/scheduler/scheduler.go [hunk 4]
```diff
@@ -78,6 +87,9 @@ func New(schedules []Schedule, opts ...OptFunc) (*Scheduler, error) {
...
+// It is a blocking operation.
```
Comment (aleksmaus) [inline internal/pkg/scheduler/scheduler.go:92]: Not sure it's worth repeating for every Run method that it is blocking. Every function is blocking by default.

File: internal/pkg/sqn/sqn.go [hunk 5]
```diff
@@ -14,13 +14,15 @@ const UndefinedSeqNo = -1
 
 var DefaultSeqNo = []int64{UndefinedSeqNo}
 
-// Abstracts the array of document seq numbers
+// SeqNo Abstracts the array of document seq numbers
```
Comment (aleksmaus) [inline internal/pkg/sqn/sqn.go:17]: nit:
"// SeqNo Abstracts the array of document seq numbers" -> "// SeqNo abstracts the array of document seq numbers"
seems like that was the convention so far

File: internal/pkg/sqn/sqn.go [hunk 6]
```diff
@@ -52,6 +54,7 @@ func (s SeqNo) toString(withBrackets bool) string {
 	return b.String()
 }
 
+// IsSet returns true when the first element in SeqNo greater than zero.
```
Comment (aleksmaus) [inline internal/pkg/sqn/sqn.go:57]: Maybe: "// IsSet returns true when the SeqNo was initialized with value."

File: internal/pkg/action/token_resolver.go [hunk 7]
```diff
@@ -17,11 +17,14 @@ import (
 
 const cacheSize = 5000
 
+// TokenResolver is an LRU cache for seqno on agent check-in
+// TODO what are the "tokens" agentID?
```
[the remark this answers is not in the document]
Comment (michel-laterman) [reply internal/pkg/action/token_resolver.go:21]: thanks, that made it a bit clearer.

File: internal/pkg/rollback/rollback.go (hunk 1 above)
Comment (michel-laterman) [reply internal/pkg/rollback/rollback.go:22]: added them everwhere

File: internal/pkg/scheduler/scheduler.go (hunk 4 above)
Comment (michel-laterman) [reply internal/pkg/scheduler/scheduler.go:92]: removed for blocking calls

**Review by aleksmaus — APPROVED** [review]
👍

**Pull request merged**